### DISCUSSION TITLE
Is SshClient threadsafe?

### DISCUSSION THREAD

**Discussion by ScuroGuardiano** [sshnet/SSH.NET#1473] [Q&A] [answered]
Is SshClient threadsafe?

**Comment by Rob-Hague** [comment, marked as answer]
It depends what you intend to do. For example, running multiple commands simultaneously should be OK, but adding/removing forwarded ports has no thread safety

**Comment by ZenithElevate** [comment]
When the ssh clients are being simultaneously instantiated and connected, for a list of hosts of which some are online, others are offline, even those that target the online hosts wait for 20 seconds to connect. This is from Task.Run(). If I remove the offline hosts from the list, then there is no delay.
